app/controllers/songs_controller.rb

```diff
@@ -12,7 +12,12 @@ def new
   end
 
   def create
+    artist = Artist.find_or_create_by(name: params[:song][:artist_name])
```

Comment:
Would be great to use `artist_name` setter method that you defined on the `Song` class.